mindmaps-graql/src/main/java/io/mindmaps/graql/internal/reasoner/predicate/Relation.java

```diff
@@ -107,6 +118,31 @@ public boolean containsVar(String name)
     }
 
     @Override
+    public MatchQueryDefault getMatchQuery(MindmapsTransaction graph)
```

Comment:
This method has a lot of duplicate code with AtomBase:: getMatchQuery(). Maybe we can push some of this code into a common accessor on the super class ?